File: files/200-299/274/CH-274.R
library(tidyverse)
library(readxl)

path = "files/200-299/274/CH-274 Custom Grouping.xlsx"
input = read_excel(path, range = "B2:C18")
test  = read_excel(path, range = "G2:I18")

result = input %>%
  mutate(Group = cumsum(c(1, diff(Date) > 2)))

all.equal(result$Group, test$Group) 
# [1] TRUE

File: files/200-299/274/CH-274.py
import pandas as pd

path = "200-299/274/CH-274 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=16)
test = pd.read_excel(path, usecols="G:I", skiprows=1, nrows=16).rename(columns=lambda col: col.replace('.1', ''))

input['Group'] = input['Date'].diff().dt.days.gt(2).cumsum() + 1

print(input.equals(test)) # True